lib/spack/llnl/util/cpu/detect.py

```diff
@@ -109,22 +110,42 @@ def sysctl(*args):
         'model name': sysctl('-n', 'machdep.cpu.brand_string')
     }
 
-    # Super hacky way to deal with slight representation differences
-    # Would be better to somehow consider these "identical"
-    if 'sse4.1' in info['flags']:
-        info['flags'] += ' sse4_1'
-    if 'sse4.2' in info['flags']:
-        info['flags'] += ' sse4_2'
-    if 'avx1.0' in info['flags']:
-        info['flags'] += ' avx'
-    if 'clfsopt' in info['flags']:
-        info['flags'] += ' clflushopt'
-    if 'xsave' in info['flags']:
-        info['flags'] += ' xsavec xsaveopt'
+    adjust_raw_flags(info)
 
     return info
 
 
+def adjust_raw_flags(info):
+    """Adjust the flags detected on the system to homogenize
+    slightly different representations.
+    """
+    # Flags detected on Darwin turned to their linux counterpart
+    flags = info.get('flags', [])
+    d2l = targets_json['miscellaneous']['darwin_flags_to_linux']
```

Comment:
I think miscellaneous is used to convert from OS detected name to standard name (used in microarchitectures).
(feature_aliases is used to convert from user input name to standard name.)

Currently standard name is same as linux.
But when linux name is changed in latest version, we will have convert table in this json object.
So I suggest to remove '_to_linux'.

Reply:
The use of the the information under `miscellaneous` and in `feature_aliases` is different. The first one is used when we get raw information from `procfs` or `sysctl` to map the raw names that have been read to the ones that correspond to our "internal" representation. The second one is a mechanism to extend the flags that a user can query on a target and has nothing to do with detection.

Concerning the naming, I wouldn't do the change since the JSON file should be readable as a standalone and naming the field "darwin_flags" will give meaning to just the keys of the object, not to the items.

Reply:
I think @tkameyama is saying that if the Linux flag names change, `to_linux` won't be accurate -- we're just using Linux as the canonical name for now.  Should we call it `to_canonical` or something?

I am not sure about the `miscellaneous` section.  Is it `miscellaneous` because it's not strictly part of detection?  I can't come up with a name off the top of my head but it seems odd to throw it all in `miscellaneous`.

Reply:
@tgamblin @tkameyama In the latest commit I renamed `miscellaneous` to `conversions` and added a `description` attribute to explain the intended use of the fields in there. With that explanation in place I think it makes sense to do as @tkameyama suggests and shorten the attribute name to `darwin_flags` - so I also did that. Let me know if there's anything else to be done here.

Reply:
It seems to good.